sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -16,17 +23,9 @@
  */
 typedef struct az_iot_hub_client_options
 {
-  /**
-   * @brief The module name (if a module identity is used).
-   *
-   */
-  az_span module_id;
-
-  /**
-   * @brief The user-agent is a formatted string that will be used for Azure IoT usage statistics.
-   *
-   */
-  az_span user_agent;
+  az_span module_id; /** @brief The module name (if a module identity is used). */
+  az_span user_agent; /** @brief The user-agent is a formatted string that will be used for Azure

```

Comment:
do we need to say that this is optional?

Reply:
We need to document the flow. According to @JeffreyRichter the way this is meant to be used is:
1. Application calls `az_iot_hub_client_options_default()`
2. Application makes changes within the `az_iot_hub_client_options` struct.

Filling in an uninitialized struct should not be supported.

Reply:
Yes, Chris is right. This is the flow. This flow eases versioning in the future. That is, we can add new fields to the options structure; the *_default() function will know about those fields and initialize ALL Fields (even the new fields). The customer can override the fields that they knew about but leave fields they don't know about as they were initialized by the _default() function. We do NOT support the pattern of a 0-initialized options structure.

For things like retry policies; using 0 for max-retries or 0 for timeouts does not work. So the _default() function initializes these fields with good defaults. This pattern also allows a customer to see our default values before passing the options to a XxxClient init function.